src/commands/moderation/Slowmode.ts

```diff
@@ -40,26 +41,35 @@ export default class SlowmodeCommand extends BaseCommand {
 	async execute(
 		client: DiscordClient<true>,
 		interaction: DiscordChatInputCommandInteraction<"cached">,
-	) {
+	): Promise<void> {
 		const timeString = interaction.options.getString("time", true);
 		const channel =
 			interaction.options.getChannel("channel", false) ??
 			interaction.channel;
 
-		const time = /^\d+$/.test(timeString)
+		const isNumber = /^\d+$/.test(timeString);
+		const time = isNumber
 			? Number.parseInt(timeString)
 			: parse(timeString, "second") ?? 0;
 
-		if (!channel || !channel.isTextBased()) {
-			interaction.reply({
+		if (!isNumber && time === 0) {
+			await interaction.reply({
+				content: "Invalid slowmode duration. Please provide a valid numerical duration (e.g., '5s', '1m', '1h').",
+				ephemeral: true,
+			});
+			return;
+		}
+
+		if (!channel || channel.type !== ChannelType.GuildText) {
+			await interaction.reply({
 				content: "Channel must be text-based",
 				ephemeral: true,
 			});
 			return;
 		}
 
 		if (time > 21600 || time < 0) {
-			interaction.reply({
+			await interaction.reply({
```

Comment:
Don't use unnecessary awaits, slows down code. Since there is a return after it it doesn't need to be awaited.